### PR TITLE
fix(libscap): increase the BPF verifier's log buffer size

### PR DESCRIPTION
**Pull request by Andreagit97**
Signed-off-by: Andrea Terzolo <andrea.terzolo@polito.it>

**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area driver-ebpf

/area libscap

**What this PR does / why we need it**:

When we load a BPF program we pass also a tmp buffer to collect the verifier logs in case of failure. If the size of the buffer is not large enough to store all verifier messages, `-1` is returned and `errno` is set to `ENOSPC`.  This is what happens on my machine (**5.11.0-1022-aws 20.04.1-Ubuntu aarch64 GNU/Linux**): 

```
bpf_load_program() err=28 event=filler/sys_autofill message=0
```
Where `28` is exactly `ENOSPC`.
This PR aligns the log buffer size to the one used by `libbpf` library, avoiding this problem. Moreover, I tried to improve our debug prints in case of verifier issues, the new output should be something like this (very similar to `libbpf`):

```
dereference of modified ctx ptr R2 off=16 disallowed
processed 8486 insns (limit 1000000) max_states_per_insn 5 total_states 151 peak_states 143 mark_read 13
-- END PROG LOAD LOG --
libscap: bpf_load_program() err=13 event=filler/sys_autofill
```
 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Wow this is super nice! Thank you!

File: userspace/libscap/scap_bpf.c [hunk 1]
```diff
@@ -59,9 +59,14 @@ struct bpf_map_data {
 
 #ifndef MINIMAL_BUILD
 
+# define UINT32_MAX (4294967295U)
```
[the remark this answers is not in the document]
Comment (Andreagit97) [reply userspace/libscap/scap_bpf.c:62]: Unfortunately, yes :(

File: userspace/libscap/scap_bpf.c [hunk 2]
```diff
@@ -450,8 +464,8 @@ static int32_t load_tracepoint(scap_t* handle, const char *event, struct bpf_ins
 	fd = bpf_load_program(prog, program_type, insns_cnt, error, BPF_LOG_SIZE);
 	if(fd < 0)
 	{
-		fprintf(stderr, "%s", error);
-		snprintf(handle->m_lasterr, SCAP_LASTERR_SIZE, "bpf_load_program() err=%d event=%s message=%s", errno, event, error);
+		fprintf(stderr, "-- BEGIN PROG LOAD LOG --\n%s-- END PROG LOAD LOG --\n", error);
```
Comment (Molter73) [inline userspace/libscap/scap_bpf.c:467]: This is a nitpick, but maybe we should place the separator on a new line to improve readability? Or are we relying on the verifier always adding a new line to the end of the errors?
```suggestion
		fprintf(stderr, "-- BEGIN PROG LOAD LOG --\n%s\n-- END PROG LOAD LOG --\n", error);
```

**Review by Molter73 — COMMENTED** [review]
Nice! This should make it a lot easier to check verifier errors!

File: userspace/libscap/scap_bpf.c (hunk 2 above)
Comment (Andreagit97) [reply userspace/libscap/scap_bpf.c]: Thank you Mauro I took it directly from `libbpf` code so it should be ok without `\n`, but just to be sure I see no problem in adding it :)

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**